codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientProtocolHandshakeHandler.java

```diff
@@ -65,4 +94,44 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             response.release();
         }
     }
+
+    public WebSocketClientProtocolHandshakeHandler handshakeTimeoutMillis(long handshakeTimeoutMillis) {
+        this.handshakeTimeoutMillis = handshakeTimeoutMillis;
+        return this;
+    }
+
+    public void applyHandshakeTimeout() {
+        final ChannelPromise localHandshakePromise = handshakePromise;
+        final long handshakeTimeoutMillis = this.handshakeTimeoutMillis;
+        if (handshakeTimeoutMillis <= 0 || localHandshakePromise.isDone()) {
```

Comment:
do we also need to check if `localHandshakePromise  == null` ?

Reply:
`handshakePromise` is initialized when the handle is added. Is there a case where `handshakePromise` is empty?